zerver/webhooks/newrelic/view.py

```diff
@@ -1,169 +1,264 @@
+
+    Returns the action taken as a string.
+    """
+    updated_at = time_stamp_map.get("UPDATED")
+    acknowledged_at = time_stamp_map.get("ACKNOWLEDGED")
+    created_at = time_stamp_map.get("CREATED")
+    closed_at = time_stamp_map.get("CLOSED")
+
+    if updated_at:
+        if acknowledged_at and updated_at == acknowledged_at:
+            return "ACKNOWLEDGED"
+        elif created_at and updated_at == created_at:
+            return "ACTIVATED"
+        elif closed_at and updated_at == closed_at:
+            return "CLOSED"
+        else:
+            return "UPDATED"
+    incident_title = payload.get("title", "No title.").tame(check_string)
+    raise UnsupportedWebhookEventTypeError(incident_title)
+
+
+def generate_additional_info_message(payload: WildValue) -> str | None:
+    """
+    Parses additional fields from the recommended base payload
+    into a presentable block of strings.
+    Detail: zerver/webhooks/newrelic/doc.md/#recommended-base-payload-template
+
+    Returns a block of formatted strings.
+    """
+    owner = payload.get("owner").tame(check_none_or(check_string))
+    additional_info_str = f"|:silhouette:  Acknowledged by **{owner}**|\n"
+    return None if owner is None else additional_info_str
+
+
+def generate_custom_field_message(payload: WildValue) -> str | None:
+    """
+    Parses the custom fields in the "zulipCustomFields" dict in
+    the payload by turning the camel case key into words and
+    formatting it together with the value into a presentable
+    block of strings.
+    Detail: zerver/webhooks/newrelic/doc.md/#displaying-custom-fields
+
+    Returns a block of formatted strings.
+    """
+    body_custom_field_detail: str = ""
+    zulip_custom_fields = payload.get(ZULIP_CUSTOM_FIELDS_KEY, {})
+
+    for key, value in zulip_custom_fields.items():
+        custom_field_name_str = key.capitalize()
+        try:
+            details = value.tame(
+                check_none_or(
+                    check_union(
+                        [
+                            check_int,
+                            check_float,
+                            check_string,
+                            check_list(
+                                check_none_or(check_union([check_int, check_float, check_string]))
+                            ),
+                        ]
+                    )
+                )
+            )
+            if isinstance(details, list):
+                custom_field_detail_str = ", ".join([f"**`{detail}`**" for detail in details])
+            else:
+                custom_field_detail_str = f"**{details}**"
+
+            custom_field_message = f"|{custom_field_name_str}: {custom_field_detail_str}|\n"
+            body_custom_field_detail += custom_field_message
+        except ValidationError:
+            invalid_field_message = f"|*The **{custom_field_name_str}** contains [unsupported field](/integrations/doc/newrelic#displaying-custom-fields) format.* |\n"
```

Comment:
I don't think we need a link here. Something simpler ...

`The value of **{custom_field_name}** is not a supported data type.`